homeassistant/components/intellidrive/cover.py

```diff
@@ -0,0 +1,153 @@
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Intellidrive covers."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            SlidingDoorCoverEntity(
+                hass,
+                coordinator,
+                entry.data[CONF_HOST],
+                str(entry.data.get(CONF_TOKEN)),
+                cast(str, entry.unique_id),
+            )
+        ]
+    )
+
+
+class SlidingDoorCoverEntity(IntelliDriveEntity, CoverEntity):
+    """Wrapper class to adapt the intellidrive device into the Homeassistant platform."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        coordinator: ReisingerCoordinator,
+        host: str,
+        token: str,
+        device_id: str,
+    ) -> None:
+        """Initialize slidingdoor cover."""
+        self._state: str | None = None
+        self._state_before_move: str | None = None
+        self._host = host
+        self._token = token
+        self._device_api = coordinator.device
+        super().__init__(coordinator, device_id)
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the door async."""
+
+        if self._state in [STATE_CLOSED, STATE_CLOSING]:
+            return
+        self._state_before_move = self._state
+        self._state = STATE_CLOSING
+
+        await self._device_api.async_close()
+
+    async def async_open_cover(self, **kwargs: Any) -> None:
+        """Open the door async."""
+
+        if self._state in [STATE_OPEN, STATE_OPENING]:
+            return
+        self._state_before_move = self._state
+        self._state = STATE_OPENING
+
+        await self._device_api.async_open()
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop the door async."""
+        await self._device_api.async_stop_door()
+
+    def stop_cover(self, **kwargs: Any) -> None:
+        """Stop the door."""
+        self.hass.async_add_executor_job(self.async_stop_cover, **kwargs)
+
+    def open_cover(self, **kwargs: Any) -> None:
+        """Open the door."""
+        self.hass.async_add_executor_job(self.async_open_cover, **kwargs)
+
+    def close_cover(self, **kwargs: Any) -> None:
+        """Close the door."""
+        self.hass.async_add_executor_job(self.async_close_cover, **kwargs)
+
+    @property
+    def device_class(self) -> CoverDeviceClass:
+        """Return the class of this device."""
+        return CoverDeviceClass.DOOR
+
+    @property
+    def supported_features(self) -> CoverEntityFeature:
+        """Flag supported features."""
+        return (
+            CoverEntityFeature.OPEN | CoverEntityFeature.CLOSE | CoverEntityFeature.STOP
+        )
+
+    @callback
+    def _update_attr(self) -> None:
+        """Update the state and attributes."""
+
+        status = self.coordinator.data
+        self._attr_name = f"Slidingdoor {status[STATUSDICT_SERIALNO]}"
```

Comment:
Don't reset the name every time you refresh the state, instead set it in `__init__` 👍